src/contract.rs

```diff
@@ -82,9 +71,9 @@ pub fn migrate(
     deps: DepsMut<ProvenanceQuery>,
     _env: Env,
     msg: MigrateMsg,
-) -> Result<Response, ContractError> {
+) -> Result<Response<ProvenanceMsg>, ContractError> {
     // Ensure that the message is valid before processing the request
     msg.validate()?;
-    let migrate_msg = msg.to_migrate_contract_v1(&deps.as_ref())?;
-    migrate_contract(deps, migrate_msg)
+    let migrate_msg = msg.to_migrate_contract_v2(&deps.as_ref())?;
+    migrate_contract(deps, &ProvenanceUtilImpl, migrate_msg)
```

Comment:
Migration will likely have this util removed once I get rid of the code that moves all the V1 stuff to V2, so I didn't bother adding a special "force the ProvenanceUtilImpl" funciton